thunder/distributed/__init__.py

```diff
@@ -594,7 +594,29 @@ def _shard_param(
         # NOTE This could be a ShardTensor to indicate other parts of the code
         # that it's sharded and should be treated differently
         shard = param.data.narrow(dim_to_shard, chunk_size * rank, chunk_size).clone()
-        param.data = shard
+        return shard, None
+
+
+def _shard_param(
```

Comment:
shouldn't this be decorated with `torch.no_grad()`?

Reply:
Great idea!

Reply:
I added the decorator to `_shard_tensor`.